params/config.go

```diff
@@ -364,91 +274,142 @@ type NodeConfig struct {
 	MailServerRegistryAddress string
 }
 
-// NewNodeConfig creates new node configuration object
-func NewNodeConfig(dataDir, clstrCfgFile, fleet string, networkID uint64) (*NodeConfig, error) {
+// NewNodeConfigWithDefaults creates new node configuration object with some defaults suitable for adhoc use
+func NewNodeConfigWithDefaults(dataDir, fleet string, networkID uint64) (*NodeConfig, error) {
+	nodeConfig, err := NewNodeConfig(dataDir, fleet, networkID)
+	if err != nil {
+		return nil, err
+	}
+
+	if dataDir != "" {
+		nodeConfig.KeyStoreDir = path.Join(dataDir, "keystore")
+		nodeConfig.WhisperConfig.DataDir = path.Join(dataDir, "wnode")
+	}
+
+	statusConfigJSON, err := static.Asset(fmt.Sprintf("config/cli/fleet-%s.json", fleet))
+	if err == nil {
+		err = LoadConfigFromJSON(string(statusConfigJSON), nodeConfig)
+	}
+	if err != nil {
+		return nil, fmt.Errorf("Default config could not be loaded: %s", err)
+	}
+
+	nodeConfig.NoDiscovery = true
+	nodeConfig.HTTPHost = ""
+	nodeConfig.ListenAddr = ":30303"
+	nodeConfig.LogEnabled = true
+	nodeConfig.LogLevel = "INFO"
+	nodeConfig.LogToStderr = LogToStderr
+	nodeConfig.WhisperConfig.EnableNTPSync = true
+
+	return nodeConfig, nil
+}
+
+// NewNodeConfig creates new node configuration object with bare-minimum defaults
+func NewNodeConfig(dataDir, fleet string, networkID uint64) (*NodeConfig, error) {
 	nodeConfig := &NodeConfig{
-		NetworkID:         networkID,
-		DataDir:           dataDir,
-		Name:              ClientIdentifier,
-		Version:           Version,
-		RPCEnabled:        RPCEnabledDefault,
-		HTTPHost:          HTTPHost,
-		HTTPPort:          HTTPPort,
-		ListenAddr:        ListenAddr,
-		APIModules:        APIModules,
-		MaxPeers:          MaxPeers,
-		MaxPendingPeers:   MaxPendingPeers,
-		IPCFile:           IPCFile,
-		log:               log.New("package", "status-go/params.NodeConfig"),
-		LogFile:           LogFile,
-		LogLevel:          LogLevel,
-		LogToStderr:       LogToStderr,
-		ClusterConfigFile: clstrCfgFile,
-		ClusterConfig: &ClusterConfig{
-			Enabled:     true, // cluster must be enabled by default
+		NetworkID:       networkID,
+		DataDir:         dataDir,
+		Version:         Version,
+		RPCEnabled:      RPCEnabledDefault,
+		HTTPHost:        HTTPHost,
+		HTTPPort:        HTTPPort,
+		ListenAddr:      ListenAddr,
+		APIModules:      APIModules,
```

Comment:
I removed about half the constants, but some are still shared with mailserver-canary and/or tests.

Reply:
We can create a follow-up PR to remove `defaults.go` completely. In my opinion, it should not exist because the usage of this package spreads everywhere.